### PR TITLE
Add & remove tracks

### PR DESCRIPTION
**Pull request by sd-owens**
- Removed orphaned gui buttons, spinboxes. Removed orphaned listeners.
- Added pop up dialog box as a user aid to load loops
- Removed obselete JSON parsing functions from ui.py
- Fixed bug in save loop that moved working directory to root incorrectly.
- added function to get relative path from absolute path in ui.py
- added remove track feature to ui.py, looper.py and connected with loop.py
- added add track feature to ui.py, looper.py and connected with loop.py
- added TODO checklist for potentially redundant and obsolete functions in looper.py
- added bug for empty tracklist loops to worklist and faulty trigger for looper to return to IDLE state if loop track list is empty.

### CODE REVIEW

File: audnauseum/state_machine/looper.py [hunk 1]
```diff
@@ -363,7 +367,15 @@ def pan_dec(self):
             return True
         return False
 
+    # TODO  This section may need cleanup, some functions are redundant or exist in seperate Track class
+    # 1. create_track does the same thing as load_track above and is implemented as append track in loop.py
```
Comment (robomoto) [inline audnauseum/state_machine/looper.py:371]: This can go.

File: audnauseum/state_machine/looper.py [hunk 2]
```diff
@@ -363,7 +367,15 @@ def pan_dec(self):
...
+    # 2. add_track is encapsulated in loop.py and is redundant here
```
Comment (robomoto) [inline audnauseum/state_machine/looper.py:372]: This can go

File: audnauseum/state_machine/looper.py [hunk 3]
```diff
@@ -363,7 +367,15 @@ def pan_dec(self):
...
+    # 3. set_track_beat_length may be obselete, this is handled in the Track class and beat_length is not a
```
Comment (robomoto) [inline audnauseum/state_machine/looper.py:373]: This can go

File: audnauseum/state_machine/looper.py [hunk 4]
```diff
@@ -363,7 +367,15 @@ def pan_dec(self):
...
+    #    parameter currently used.
+    # 4. calc_track_bpm is accomplished in object creation in Track?  Redundant?
```
Comment (robomoto) [inline audnauseum/state_machine/looper.py:375]: Yeah, that's old code.  That can go.

File: resources/json/cd_test2.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
     "__type__": "Loop",
-    "file_path": "resources/json/cd_test2.json",
+    "file_path": "/Users/steve/Projects/Python/AudNauseum/resources/json/cd_test2.json",
```
Comment (robomoto) [inline resources/json/cd_test2.json:3]: I would not upload this change.

**Review by robomoto — CHANGES_REQUESTED** [review]
I would just remove your change to cd_test2.json, which only adds a user-specific directory
Everything else looks great.  Your call-outs of things that can go away all look good.

File: resources/json/cd_test2.json (hunk 5 above)
Comment (sd-owens) [reply resources/json/cd_test2.json]: see commit comments below

File: audnauseum/state_machine/looper.py (hunk 4 above)
Comment (sd-owens) [reply audnauseum/state_machine/looper.py]: see commit comments below

File: audnauseum/state_machine/looper.py (hunk 3 above)
Comment (sd-owens) [reply audnauseum/state_machine/looper.py]: see commit comments below

File: audnauseum/state_machine/looper.py (hunk 2 above)
Comment (sd-owens) [reply audnauseum/state_machine/looper.py]: see commit comments below

File: audnauseum/state_machine/looper.py (hunk 1 above)
Comment (sd-owens) [reply audnauseum/state_machine/looper.py]: see commit comments below

**Pull request merged**